TYPO3.Neos/Classes/TYPO3/Neos/TypoScript/VariantMenuImplementation.php

```diff
@@ -74,17 +75,22 @@ protected function buildItems()
                 $nodeInDimensions = null;
             }
 
-            // determine labels for target dimensions of node
-            array_walk($targetDimensions, function (&$dimensionValue, $dimensionName, $allDimensionPresets) {
-                $dimensionValue = ['value' => $dimensionValue, 'label' => $allDimensionPresets[$dimensionName]['presets'][$dimensionValue]['label']];
+            // determine metadata for target dimensions of node
+            array_walk($targetDimensions, function (&$dimensionValue, $dimensionName, $allDimensionPresets) use ($pinnedDimensionName) {
+                $dimensionValue = [
+                    'value' => $dimensionValue,
+                    'label' => $allDimensionPresets[$dimensionName]['presets'][$dimensionValue]['label'],
+                    'isPinnedDimension' => ($pinnedDimensionName === null || $dimensionName == $pinnedDimensionName) ? true : false
+                ];
             }, $allDimensionPresets);
 
             $menuItems[] = [
                 'node' => $nodeInDimensions,
                 'state' => $this->calculateItemState($nodeInDimensions),
                 'label' => $nodeInDimensions === null ? '' : $nodeInDimensions->getLabel(),
                 'dimensions' => $allowedCombination,
-                'targetDimensions' => $targetDimensions
+                'targetDimensions' => $targetDimensions,
+                'pinnedDimensionName' => $pinnedDimensionName
```

Comment:
As the pinned dimension is the same for all menu entries I don't see a use for it in this array, if you need it you can directly access the value in the template...